discover_modes emits DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.

Saw this in astrowidgets dev testing (https://github.com/astropy/astrowidgets/pull/156). Also see https://github.com/python/importlib_metadata/issues/298 . I am going to ignore this warning downstream but you might want to consider addressing it here.

Example log: https://github.com/astropy/astrowidgets/actions/runs/3659056477/jobs/6184616525

```
>       iw = ImageWidget()

../../astrowidgets/tests/test_image_widget.py:268: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../astrowidgets/core.py:76: in __init__
    self._viewer = EnhancedCanvasView(logger=logger)
.../ginga/web/jupyterw/ImageViewJpw.py:417: in __init__
    ImageViewZoom.__init__(self, logger=logger, settings=settings,
.../ginga/web/jupyterw/ImageViewJpw.py:400: in __init__
    self.set_bindings(bindings)
.../ginga/web/jupyterw/ImageViewJpw.py:410: in set_bindings
    bindings.set_bindings(self)
.../ginga/Bindings.py:89: in set_bindings
    self.setup_settings_events(viewer, bindmap)
.../ginga/Bindings.py:157: in setup_settings_events
    from ginga.modes.modeinfo import available_modes
.../ginga/modes/modeinfo.py:60: in <module>
    discover_modes()
.../ginga/modes/modeinfo.py:30: in discover_modes
    discovered_modes = entry_points().get(group, [])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = {'array_api': [EntryPoint(name='numpy', value='numpy.array_api', group='array_api')],
        'asdf_extensions': [EntryPoint(n..._finalize', value='setuptools.dist:_Distribution.finalize_options',
                                       group='setuptools.finalize_distribution_options')]}
name = 'ginga_modes', default = []

    def get(self, name, default=None):
>       self._warn()
E       DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.

.../python3.11/importlib/metadata/__init__.py:443: DeprecationWarning
```

--- a/doc/WhatsNew.rst
+++ b/doc/WhatsNew.rst
@@ -37,6 +37,7 @@ Ver 4.0.0 (unreleased)
 - Fixed an issue where manual cut levels changes weren't reflected in
   the Thumbs icon
 - ScreenShot now correctly captures the background color of the viewer
+- Fixed a DeprecationWarning related to use of entry points
 
 Ver 3.4.0 (2022-06-28)
 ======================

--- a/ginga/modes/modeinfo.py
+++ b/ginga/modes/modeinfo.py
@@ -2,17 +2,11 @@
 # This is open-source software licensed under a BSD license.
 # Please see the file LICENSE.txt for details.
 #
-import sys
 import os
 import glob
 import inspect
-if sys.version_info < (3, 8):
-    # Python 3.7
-    from importlib_metadata import entry_points
-else:
-    from importlib.metadata import entry_points
 
-from ginga.util import paths
+from ginga.util import paths, compat
 from ginga.modes.mode_base import Mode
 from ginga.misc.ModuleManager import my_import
 
@@ -27,7 +21,7 @@ def add_mode(mode_class):
 def discover_modes():
     # discover modes installed under entry point "ginga_modes"
     group = 'ginga_modes'
-    discovered_modes = entry_points().get(group, [])
+    discovered_modes = compat.ep_get(group)
     for entry_point in discovered_modes:
         try:
             mode_class = entry_point.load()

--- a/ginga/rv/main.py
+++ b/ginga/rv/main.py
@@ -15,18 +15,12 @@ import logging.handlers
 import threading
 import traceback
 
-if sys.version_info < (3, 8):
-    # Python 3.7
-    from importlib_metadata import entry_points
-else:
-    from importlib.metadata import entry_points
-
 # Local application imports
 from ginga.misc.Bunch import Bunch
 from ginga.misc import Task, ModuleManager, Settings, log
 import ginga.version as version
 import ginga.toolkit as ginga_toolkit
-from ginga.util import paths, rgb_cms, json
+from ginga.util import paths, rgb_cms, json, compat
 
 # Catch warnings
 logging.captureWarnings(True)
@@ -211,7 +205,7 @@ class ReferenceViewer(object):
         available_methods = []
 
         for group in groups:
-            discovered_plugins = entry_points().get(group, [])
+            discovered_plugins = compat.ep_get(group)
             for entry_point in discovered_plugins:
                 try:
                     method = entry_point.load()

--- a/ginga/util/compat.py
+++ b/ginga/util/compat.py
@@ -1,0 +1,21 @@
+#
+# compat.py -- Python compatibility shims.
+#
+# This is open-source software licensed under a BSD license.
+# Please see the file LICENSE.txt for details.
+#
+import sys
+if sys.version_info < (3, 8):
+    # Python 3.7
+    from importlib_metadata import entry_points
+else:
+    from importlib.metadata import entry_points
+
+
+def ep_get(group):
+    """Get the entry points for a group."""
+    ep = entry_points()
+    if hasattr(ep, "select"):
+        return ep.select(group=group)
+    else:
+        return ep.get(group, [])

